File: 2025/day06/Python/part1/day06pt1.py
def main():
    with open("input.txt", "r", encoding="utf-8") as file:
        nums = []
        for line in file:
            split = line.split()
            nums.append(split)

        operations = nums[-1]
        nums = nums[:-1]

        cols = len(operations)
        res = 0

        for col in range(cols):
            col_vals = []
            for row in range(len(nums)):
                col_vals.append(int(nums[row][col]))

            if operations[col] == "+":
                res += sum(col_vals)
            else:
                holder = 1
                for vals in col_vals:
                    holder *= vals
                
                res += holder

    print(res)

main()



        

File: 2025/day06/Go/day06pt1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
		return
	}
	defer file.Close()

	var nums [][]string

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		split := strings.Fields(line)


		nums = append(nums, split)
	}

	if err := scanner.Err(); err != nil {
		panic(err)
		return
	}

	operations := nums[len(nums)-1]
	nums = nums[:len(nums)-1]

	cols := len(operations)
	res := 0

	for col := 0; col < cols; col++ {
		colVals := make([]int, 0, len(nums))

		for row := 0; row < len(nums); row++ {

			val, err := strconv.Atoi(nums[row][col])
			if err != nil {
				panic(err)
				return
			}

			colVals = append(colVals, val)
		}

		if operations[col] == "+" {
			sum := 0
			for i := 0; i < len(colVals); i++ {
				sum += colVals[i]
			}
			res += sum
		} else {
			holder := 1
			for i := 0; i < len(colVals); i++ {
				holder *= colVals[i]
			}
			res += holder
		}
	}

	fmt.Println(res)
}